storage/objects/generate_encryption_key.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objects
+
+// [START storage_generate_encryption_key]
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+)
+
+// generateEncryptionKey generates a base64-encoded encryption key.
+func generateEncryptionKey(w io.Writer) error {
```

Comment:
I notice that the C++ sample here has a lot more information about encryption. Do we want to include that in the other samples (or pull up into the docs)? @frankyn  https://cloud.google.com/storage/docs/encryption/using-customer-supplied-keys#generating_your_own_encryption_key

Reply:
The C++ sample seems a little excessive to me, but some other samples have a little piece of clarification text that I think should be included here:
```
    Generates a 256 bit (32 byte) AES encryption key and prints the
    base64 representation.

    This is included for demonstration purposes. You should generate your own
    key. Please remember that encryption keys should be handled with a
    comprehensive security policy.
```

Reply:
Done